### DISCUSSION TITLE
Non impacting upgrades

### DISCUSSION THREAD

**Discussion by andyc1864** [keycloak/keycloak#14542] [Ideas]
Hi everyone.

I have a situation where I need to implement upgrades of services without impacting the end users. An approach we are using for a number of services is Blue/Green using separate namespaces in a Kubernetes cluster.

I need to cover off version upgrades for Keycloak and introducing them in a non-impacting way. Is there a process /method that anyone can help me with? I have read the upgrade documentation, but they all seem to depend on a cutover point which would have an impact on the end user.

We are currently running Keycloak in a Kubernetes cluster, and are using Infinispan.

Thanks for your time.

Andy

We

**Comment by sschu** [comment]
Keycloak currently does not support zero-downtime upgrades. With the new map storage, this will be supported.

**Reply by andyc1864** [reply]
Hi, thanks for your prompt reply. Do you know when this feature will be available,  so I can try and factor into my plans?

Best wishes.

Andy C

**Reply by sschu** [reply]
It should be feature complete with KC20. I guess it will take some time to really make it production-ready, so I would probably wait for KC21/22.

**Comment by fotag** [comment]
Hello everyone!

I apologize in advance if this is not the right place to ask my question, but I believe it is related to my investigation.

We have started building our platform's authentication and authorization around Keycloak. As far as I understand, Keycloak releases updates frequently, so it would be nice if we could upgrade our production Keycloak cluster as quickly as possible. Like @andyc1864, we do not want to impact users during the upgrade process. However, we are hosting Keycloak on AWS ECS Fargate instead of Kubernetes and with AWS Aurora Postgres database.

It would be very helpful if someone could point me to any related documentation or Github issues to watch for related functionality!

What I've already found:
* https://www.keycloak.org/2022/07/storage-map.html
* https://github.com/keycloak/keycloak/issues/14455

Thank you in advance!
